packages/python/plotly/plotly/express/_core.py

```diff
@@ -813,6 +813,12 @@ def build_dataframe(args, attrables, array_attrables):
     array_attrables : list
         argument names corresponding to iterables, such as `hover_data`, ...
     """
+    args = dict(input_args)
```

Comment:
I don't think this makes a deep copy...

```
>>> x= [1,2,3]
>>> y = dict(a=x)
>>> z = dict(y)
>>> z["a"] is x
True
>>> z["a"][0]=100
>>> z
{'a': [100, 2, 3]}
>>> x
[100, 2, 3]
```

But also think this is only an issue for `array_attrables` where we write into `args[something][something]` so if we just copy those that's enough:

```
>>> x = [1,2,3]
>>> y = list(x)
>>> y is x
False
>>> y[0] = 100
>>> x
[1, 2, 3]
```

Reply:
oh no! I knew it does a deep copy for a list so I had wrongly assumed I could transpose this pattern to a `dict`.